src/wp-includes/class-wp-autoload.php

```diff
@@ -0,0 +1,516 @@
+<?php
+/**
+ * Autoloader for WordPress classes.
+ *
+ * Include this file if you'd like to avoid having to create your own autoloader.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Autoloader.
+ */
+final class WP_Autoload {
+
+	/**
+	 * An array of classes and their respective paths.
+	 *
+	 * Note: The classnames should be lowercased.
+	 *
+	 * @access private
+	 *
+	 * @var array
+	 */
+	const CLASSES_PATHS = array(
```

Comment:
There will need to be a unit test to ensure all the keys are lower case given the `strtolower()` below.

Use a data provider so if multiple classes are added with an incorrect case they all get caught on a single test run.

Reply:
@peterwilsoncc good call!
I added some basic tests in https://github.com/WordPress/wordpress-develop/pull/3470/commits/9e181d6b161d9d08b6ec4d8716a8d6f51fa5c941
These tests check the classmap and confirm that:
* Keys in the array are lowercase
* Files in the values exist
* Files defined include the corresponding class